### DISCUSSION TITLE
Why p10k has a loading period, unlike other themes?

### DISCUSSION THREAD

**Discussion by OscarGongSAP** [marlonrichert/zsh-snap#198] [Q&A] [answered]
Why zsnap with p10k has a loading period, unlike other themes, such as `sindresorhus/pure`, starship?

Here is my `.zshrc`:
```
# --------- Znap! -----------
source ~/opt/zsh-snap/znap.zsh
znap source zsh-users/zsh-autosuggestions
znap source zsh-users/zsh-syntax-highlighting
znap prompt romkatv/powerlevel10k

# To customize prompt, run `p10k configure` or edit ~/.p10k.zsh.
[[ ! -f ~/.p10k.zsh ]] || source ~/.p10k.zsh
```

https://user-images.githubusercontent.com/108839639/196109948-961d1321-e1f8-45ab-8663-e3e8e5d79ed6.mov

**Comment by marlonrichert** [comment, marked as answer]
P10k is not compatible with `znap prompt`, because it already does something similar to reduce your shell's startup time. Use `znap source` instead.

I added a note about it to the [example `.zshrc`](https://github.com/marlonrichert/zsh-snap/blob/90df35bf052f5de374132e3531bf7b61c974af7b/.zshrc#L33).

**Reply by OscarGongSAP** [reply]
HI @marlonrichert, thanks so much of your answer!
